src/sas/qtgui/Plotting/Slicers/BaseInteractor.py

```diff
@@ -59,45 +59,44 @@ def clear_markers(self):
         """
         Clear old markers and interfaces.
         """
-        for h in self.markers:
-            h.remove()
-        if self.markers:
-            self.base.connect.clear(*self.markers)
-        self.markers = []
+        try:
+            for h in self.markers:
+                h.remove()
+            if self.markers:
+                self.base.connect.clear(*self.markers)
+            self.markers = []
```

Comment:
You probably want to keep deleting the markers on exception. Currently, the first exception will stop subsequent removals. Consider wrapping just `h.remove()` with try..except